### PR TITLE
Package as npm module only

### PR DESCRIPTION
**Pull request by jemmyw**
The ruby part of this code doesn't really do anything. This PR removes the rails engine and just packages as a straight npm package react component.

Includes moving the auth button into react code and wrapping loading of the gapi javascript. Moving the auth button has the additional advantage that it now lets you reauthenticate while keeping your existing searches if your token expires.

See https://github.com/aha-app/aha-app/pull/6277

### CODE REVIEW

File: app/assets/javascripts/big_query_log_viewer/utils/query.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
+// @ts-check
```
Comment (jsteel) [inline app/assets/javascripts/big_query_log_viewer/utils/query.js]: Should we be checking in typescript stuff, or is this something else?

**Comment by jsteel** [comment]
screen shot 2018-06-28 at 8 38 58 am
screen shot 2018-06-28 at 8 39 11 am

File: app/assets/javascripts/big_query_log_viewer/utils/query.js (hunk 1 above)
Comment (jemmyw) [reply app/assets/javascripts/big_query_log_viewer/utils/query.js]: editor hint. removed

File: lib/generators/templates/big_query_log_viewer.rb [hunk 2]
```diff
@@ -1,26 +0,0 @@
-# Configuration settings for BigQueryLogViewer.
-
-# Your client ID from the Google developer console.
-BigQueryLogViewer.client_id = ''
-
-# Your project number from the Google developer console.
-BigQueryLogViewer.project_number = ''
-
-# The prefix of your table.
```
Comment (jsteel) [inline lib/generators/templates/big_query_log_viewer.rb]: Would you mind moving these descriptions over to the component?

Comment (jemmyw) [reply lib/generators/templates/big_query_log_viewer.rb]: @jsteel done and readme file also updated

File: src/utils/query.js [hunk 3]
```diff
@@ -253,10 +263,25 @@ export default class Query {
         'Starting RESQUE from "from_request_uuid"=>"abcdef-hijklmnop-qrstuv-wxyz"',
       ];
 
-      return {
+      /** @type {TestClient} */
+      const client = {
         query(args) {
           return {
             execute(callback) {
+              console.log('execute');
```
Comment (schneidmaster) [inline src/utils/query.js]: these seem like stray `console.log`s fwiw

Comment (jemmyw) [reply src/utils/query.js]: they're in the test client so probably want to stay, I'll mark them

**Pull request merged**